File: Cs/Solution/GetSneakyNumbers.cs
namespace Solution
{
	public partial class Solution
	{
		public int[] GetSneakyNumbers(int[] nums)
		{
			int[] count = new int[nums.Length];
			int[] ret = new int[2];
			int ret_count = 0;
			for (int i = 0; i < nums.Length; i++)
			{
				if (count[nums[i]] == 1)
				{
					ret[ret_count++] = nums[i];
				}
				count[nums[i]]++;
			}
			return ret;
		}
	}
}

File: Rust/src/solution/get_sneaky_numbers.rs
use super::Solution;

impl Solution {
    pub fn get_sneaky_numbers(nums: Vec<i32>) -> Vec<i32> {
        let mut sneaky_numbers = Vec::new();
        let mut cnt = vec![0; nums.len() - 2];
        let mut index = 0;
        for &num in &nums {
            cnt[num as usize] += 1;
            if cnt[num as usize] == 2 {
                sneaky_numbers.push(num);
                index += 1;
                if index == 2 {
                    break;
                }
            }
        }
        sneaky_numbers
    }
}
